libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1729,31 +1730,67 @@ bool TypeChecker::visit(UnaryOperation const& _operation)
 		requireLValue(_operation.subExpression(), false);
 	else
 		_operation.subExpression().accept(*this);
-	Type const* subExprType = type(_operation.subExpression());
-	TypeResult result = type(_operation.subExpression())->unaryOperatorResult(op);
-	if (!result)
+	Type const* operandType = type(_operation.subExpression());
+
+	// Check if the operator is built-in or user-defined.
+	TypeResult builtinResult = operandType->unaryOperatorResult(op);
+	set<FunctionDefinition const*, ASTNode::CompareByID> matchingDefinitions = operandType->operatorDefinitions(
+		op,
+		*currentDefinitionScope(),
+		true // _unary
+	);
+
+	// Operator can't be both user-defined and built-in at the same time.
+	solAssert(!builtinResult || matchingDefinitions.empty());
```

Comment:
But they have to be *either* user-defined *or* built-in, don't they :-)?
I.e.

```suggestion
	solAssert(!builtinResult != matchingDefinitions.empty());
```
would be the better assertion, wouldn't it :-)?

Reply:
Yeah, that's an explicit error case below... might be easier to read to put this closer together with that below, but also fine.

Reply:
Well, it's pretty much next to the `if`. I could put the `else` case first but not sure it actually improves things. It then moves other things away :)